cirq-core/cirq/transformers/measurement_transformers.py

```diff
@@ -111,23 +110,22 @@ def defer(op: 'cirq.Operation', _) -> 'cirq.OP_TREE':
         elif protocols.is_measurement(op):
             return [defer(op, None) for op in protocols.decompose_once(op)]
         elif op.classical_controls:
-            controls = []
+            new_op = op.without_classical_controls()
             for c in op.classical_controls:
                 if isinstance(c, value.KeyCondition):
                     if c.key not in measurement_qubits:
                         raise ValueError(f'Deferred measurement for key={c.key} not found.')
-                    qubits = measurement_qubits[c.key]
-                    if len(qubits) != 1:
-                        # TODO: Multi-qubit conditions require
-                        # https://github.com/quantumlib/Cirq/issues/4512
-                        # Remember to update docstring above once this works.
-                        raise ValueError('Only single qubit conditions are allowed.')
-                    controls.extend(qubits)
+                    qs = measurement_qubits[c.key]
+                    if len(qs) == 1:
+                        control_values: Any = range(1, qs[0].dimension)
+                    else:
+                        all_values = itertools.product(*[range(q.dimension) for q in qs])
+                        anything_but_all_zeros = tuple(itertools.islice(all_values, 1, None))
+                        control_values = ops.SumOfProducts(anything_but_all_zeros)
```

Comment:
```suggestion
                    all_values = itertools.product(*[range(q.dimension) for q in qs])
                    anything_but_all_zeros = tuple(itertools.islice(all_values, 1, None))
                    control_values = ops.SumOfProducts(anything_but_all_zeros)
```

Reply:
Logically yes, but for a single qubit, the range gets read as a ProductOfSums, which prints nicer circuit diagrams. (Last I checked SumOfProducts failed to print at all).

Reply:
After https://github.com/quantumlib/Cirq/pull/5788 is merged; `SumOfProducts` will also have nice circuit diagrams.

Reply:
Hmm, I think this is fine.